### PR TITLE
Ensure that if the preference change the PB panel gets updated.

### PR DESCRIPTION
**Pull request by MartinImgSpc**

### CODE REVIEW

File: Editor/EditorCore/ProBuilderEditor.cs [hunk 1]
```diff
@@ -499,16 +502,21 @@ void OnGUI()
             }
         }
 
-        void Menu_ToggleIconMode()
+        void IconModeChanged()
         {
-            s_IsIconGui.value = !s_IsIconGui.value;
             if (s_EditorToolbar != null)
                 DestroyImmediate(s_EditorToolbar);
             s_EditorToolbar = ScriptableObject.CreateInstance<EditorToolbar>();
```
Comment (thomas-tu) [inline Editor/EditorCore/ProBuilderEditor.cs]: I would move these three lines into one "OpenEditor()" definition as we have the same scope in InitGUI().

**Comment by thomas-tu** [comment]
Would it be possible to have a test or two for this please? :) Thanks!

**Comment by MartinImgSpc** [comment]
> Would it be possible to have a test or two for this please? :) Thanks!

With visual diff of the screenshot of the Probuilder Window?

File: Editor/EditorCore/EditorToolbar.cs [hunk 2]
```diff
@@ -17,7 +17,7 @@ sealed class EditorToolbar : ScriptableObject
         public EditorWindow window;
 
         bool isFloating { get { return ProBuilderEditor.instance != null && ProBuilderEditor.instance.isFloatingWindow; } }
-        bool isIconMode = true;
+        public bool isIconMode = true;
```
[the remark this answers is not in the document]
Comment (MartinImgSpc) [reply Editor/EditorCore/EditorToolbar.cs]: Changing

File: Editor/EditorCore/ProBuilderEditor.cs (hunk 1 above)
Comment (MartinImgSpc) [reply Editor/EditorCore/ProBuilderEditor.cs]: Changing

**Comment by gabrielw-us** [comment]
Just pulled this and opened my project, many type/namespace errors related to unit tests, perhaps it's not ready yet? Should I wait?

**Comment by MartinImgSpc** [comment]
> Just pulled this and opened my project, many type/namespace errors related to unit tests, perhaps it's not ready yet? Should I wait?

I am trying to figure out the dependency for unit tests. You can wait until all checks are successful. Thank you!

**Comment by thomas-tu** [comment]
> > Would it be possible to have a test or two for this please? :) Thanks!
> 
> With visual diff of the screenshot of the Probuilder Window?

As said to Martin on slack (I'm putting it here to everyone): covering the code part would be enough (we can set Pref values from code).
Testing the changes from the Preferences window to code is more a job for the settings-manager itself.

File: Editor/EditorCore/ProBuilderEditor.cs [hunk 3]
```diff
@@ -449,6 +443,9 @@ public static void Refresh(bool vertexCountChanged = true)
 
         void OnGUI()
         {
+            if (s_EditorToolbar != null && s_EditorToolbar.isIconMode != s_IsIconGui.value)
```
[the remark this answers is not in the document]
Comment (MartinImgSpc) [reply Editor/EditorCore/ProBuilderEditor.cs]: Is there a way to get a notification on preference changes? From what I gatter it does not seem to be the case, so we need polling. Or am I missing your point?

**Pull request merged**